### ISSUE TITLE
Windows compile issue

### ISSUE THREAD

**Issue by aalavandhaann** [RosettaCommons/binder#31]
Dear Wizards,

I was able to use Binder successfully and produce a working python library for Linux boxes. The link to my project is [here](https://github.com/aalavandhaann/chenhancc). Now I am struggling to get the build process work with Windows machines. I have access to a windows 10 machine and that's where my testing is done for windows using Visual Studio 2017. I can say that my library is minimal and does not depend on any external libraries. 

When I try to run ```python setup.py build``` I get the errors as seen in the attachment
[winbuild.log](https://github.com/RosettaCommons/binder/files/1353741/winbuild.log). The primary error is about ```error C2039: '_Rb_tree_const_iterator': is not a member of 'std'```. Am not sure fixing this will fix other errors as well. 

Anyone any ideas where things might be going wrong?

Regards,

#0K

**Comment by lyskov** [comment]
I do not have Windows machine in my possession so only little i can do to help you with this issue.

From the error that you posted it looks like you trying to compile bindings that was generated on Unix platform. If this is so then such approach will not work due to incompatibility of libstdc++ between Windows and Unix. I see two way to solve this: 
1. Generate bindings on Windows platform using MSVC std lib
2. Use Win10 Ubuntu Linux layer to run your code. That way you can compile native Linux code on Windows without modification and even re-use Linux binaries without recompiling them. You can sind some info of how to install Win10 Linux layer here: http://www.pyrosetta.org/documentation/windows

**Comment by aalavandhaann** [comment]
So it is necessary to generate the bindings as well for each platform? Sorry for having missed this point. I will for sure generate the bindings for windows and then try this again. 

The Win10 Ubuntu Linux layer to run unix code looks promising as well. I will investigate further and see which suits better. 

Regards,

#0K

**Comment by lyskov** [comment]
@aalavandhaann let me rephrase this:
> So it is necessary to generate the bindings as well for each platform?
 - if you going to create bindings for any `std::` classes then yes, it might be necessary to generate bindings on a target platform. But, 
 - if you going to generate bindings _only_ for your types or for types which definition does not change between current machine and target system then it is perfectly ok to cross-generate bindings (or parts of the bindings).

(I saw `_Rb_tree_const_iterator` in your output so it must mean that you allowed Binder to generate bindings for some types from GNU libstdc++)

**Comment by lyskov** [comment]
@aalavandhaann i think it is clear now what was the reason for errors so i am going to close this now.
